src/firewall/core/io/zone.py

```diff
@@ -199,6 +199,11 @@ def _check_config(self, config, item, all_config):
                 if not checkIPnMask(source) and not checkIP6nMask(source) and \
                    not check_mac(source) and not source.startswith("ipset:"):
                     raise FirewallError(errors.INVALID_ADDR, source)
+        elif item == "destination":
+            for destination in config:
+                if not checkIPnMask(destination) and not checkIP6nMask(destination) and \
+                   not destination.startswith("ipset:"):
+                    raise FirewallError(errors.INVALID_ADDR, destination)
```

Comment:
This should actually be `INVALID_DESTINATION`.

Reply:
Actually this whole block is not needed. This part of `check_config` applies to zone based elements. Not rich rules.